public/app/features/query/components/QueryEditorRows.tsx

```diff
@@ -11,6 +11,7 @@ import {
   PanelData,
 } from '@grafana/data';
 import { getDataSourceSrv, reportInteraction } from '@grafana/runtime';
+import { ContentOutlineItem } from 'app/features/explore/ContentOutline/ContentOutlineItem';
```

Comment:
This is breaking the Acyclic Dependency Principle and might be risky to add (Explore depends on QueryRows, and QueryRows depends on Explore).

A way around this could be:
* (1) moving ContentOutline to `features/`:  this is the long-run plan anyway but it's not very scalable/flexible: consumers have it always added, with no way to control it. `QueryEditorRow` is a generic component, seems to me it should not know the `ContentOutline` at all.
* (2) alternatively we could apply DIP and pass a Wrapper HOC to `QueryEditorRows` component for the QueryRow to create `<ContentOutlineItem.../>` in Explore. A drawback would be it could be easily abused by other consumers and used to create some additional UI which may lead to inconsistencies in the UI.
* (3) another option that could be cool (but probably more complex to properly clean up) is to register content outline items programmatically based on only `ref`, e.g.:

```typescript
<SomeExternalComponent ref={ref => contentOutlineContext.registerItem(ref, "myId", ...) />
```

Then apply the same principle to QueryEditorRow/s.

What do you think?